File: sdk/ts/src/stats.ts
import { ClientCore } from "./core.js";
import type { StatsResponse } from "./types.js";

export class StatsClient {
  constructor(private readonly core: ClientCore) {}

  /** Get runtime statistics. */
  async get(): Promise<StatsResponse> {
    return this.core.doJSON<StatsResponse>("GET", "/v1/stats");
  }
}

File: sdk/filegate/stats.go
package filegate

import (
	"context"
	"net/http"
)

// StatsClient contains statistics endpoints.
type StatsClient struct {
	core *clientCore
}

func (c StatsClient) Get(ctx context.Context) (*StatsResponse, error) {
	var out StatsResponse
	if err := c.core.doJSON(ctx, http.MethodGet, "/v1/stats", nil, nil, "", &out); err != nil {
		return nil, err
	}
	return &out, nil
}
